cmd/mt-parrot/generate.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"time"
+
+	"github.com/grafana/metrictank/schema"
+)
+
+func produceTestMetrics(schemas []*schema.MetricData) {
+	for tick := range time.NewTicker(testMetricsInterval).C {
+		for _, metric := range schemas {
+			metric.Time = tick.Unix()
+			metric.Value = float64(tick.Unix())
+		}
+		publisher.Flush(schemas)
```

Comment:
my advice:
* after Flush(), use an atomic set of a shared variable that you set to tick.Unix().  the monitor routine then knows what the last flushed timestamp is (by doing an atomic read), and thus it can determine the lag
* note that the publisher has an internal queue so the data isn't actually flushed after this call returns. we may want to add a synchronous publishing method (such that Flush only returns after the data has actually been accepted by the gateway), but for now this should be negligible.

Reply:
18918d1fe6faae2776d1b6229d0eac5b41eb4a30